addons/account/wizard/account_merge_wizard.py

```diff
@@ -0,0 +1,319 @@
+from odoo import _, api, fields, models, Command
+from odoo.exceptions import UserError
+from odoo.tools import SQL
+
+
+class AccountMergeWizard(models.TransientModel):
+    _name = 'account.merge.wizard'
+    _description = "Account merge wizard"
+
+    account_ids = fields.Many2many('account.account')
+    is_group_by_name = fields.Boolean(
+        string="Group by name?",
+        default=True,
+        help="Tick this checkbox if you want accounts to be grouped by name for merging."
+    )
+    wizard_line_ids = fields.One2many(
+        comodel_name='account.merge.wizard.line',
+        inverse_name='wizard_id',
+        compute='_compute_wizard_line_ids',
+        store=True,
+        readonly=False,
+    )
+    disable_merge_button = fields.Boolean(compute='_compute_disable_merge_button')
+
+    @api.model
+    def default_get(self, fields):
+        res = super().default_get(fields)
+        if not set(fields) & {'account_ids', 'wizard_line_ids'}:
+            return res
+
+        if self.env.context.get('active_model') != 'account.account' or not self.env.context.get('active_ids'):
+            raise UserError(_("This can only be used on accounts."))
+
+        res['account_ids'] = [Command.set(self.env.context.get('active_ids'))]
+        return res
+
+    @api.model
+    def _get_grouping_key(self, account, group_by_name=False):
```

Comment:
Given that the client is unaware of the grouping, we do the grouping server-side before creating the wizard lines, in order to create the wizard lines in the right order. So the grouping key is a function of the account, rather than of the wizard line.

I'll remove the `self.ensure_one()`.